src/rgw/setup/rgw.py

```diff
@@ -852,6 +856,82 @@ def _verify_backend_store_value(conf: MappedConf):
                 f'Supported rgw backend store are {const.SUPPORTED_BACKEND_STORES},'
                 f' currently configured one is {backend_store}')
 
+    @staticmethod
+    def _validate_resource_limit_values(conf: MappedConf):
+        """Validate minimum values of rgw resource limits in Gconf."""
+        Log.info(f'Validating minimum values of resource limits for {const.COMPONENT_NAME}.')
+        num_services = int(Rgw._get_cortx_conf(conf, const.SVC_LIMIT_NUM_SERVICES))
+        if num_services == 0:
+            raise SetupError(errno.EINVAL,
+                f'Invalid/Missing values found in gconf for key :{const.SVC_LIMIT_NUM_SERVICES}')
+        input_cpu_min_val = ''
+        input_mem_min_val = ''
+        for value_index in range(0, num_services):
```

Comment:
I really dont like this  looping over, to find rgw service resource limits. Doesnt conf store give a way to make it available as dictionary, where on feeding 'rgw' key, we get the values ?
latest solution.yaml https://github.com/Seagate/cortx-k8s/blob/v0.9.0/k8_cortx_cloud/solution.example.yaml

Reply:
As of now, consul based confstore does not provide any way to return any list.
hence we need to fetch num_services key and then iterate over each keys using count(value_index) variable and then fetch actual value.

Regarding latest gconf keys in solution.yaml..
below is updated copy which is being used in deployment 
( https://seagate-systems.atlassian.net/wiki/spaces/PUB/pages/651132980/CORTX+GConf+Config+Store+Keys#Example%3A-CORTX-GConf-Part-A-(%2Fetc%2Fcortx%2Fcluster.conf) Line : 176 -185

Today we have only one service (service : rgw) for which this limits are defined but there can be multiple services for which limit will be added. hence to differentiate between rgw and other service, I have added a check on service name parameter while iterating over idex.

Reply:
alright